src/backend/access/transam/walinsertlock.c

```diff
@@ -0,0 +1,132 @@
+#include "walinsertlock.h"
+#include "storage/proc.h"
+#include "c.h"
```

Comment:
nit: The usual convention is to have `#include "postgres.h"` as the first include in the file, followed by any system headers, followed by any other Postgres headers. All in alphabetical order. (see other files for precedent)